docs/root/overview.rst

```diff
@@ -0,0 +1,97 @@
+
+For H3, it is anticipated that it will fit into this model, but if all else fails, it will be entirely possible to wire in a new type of **BenchmarkClient.**
+
+### RequestSource
+
+**RequestSource **is an abstraction that allows us to implement different ways for **BenchmarkClient** to get information on what the request that it is about to fire off should look like. Today, two implementations exist: a static one, which will repeat the same request over and over, as well as one that pulls dynamic request data from a grpc service. The latter can, for example, be used to implement log-replay.
+
+### StreamDecoder
+
+StreamDecoder is a Nighthawk-specific implementation of an Envoy concept. It is responsible for coordinating lifetime events of a request to upper abstraction layers (**BenchmarkClient**, **Sequencer) **as well as latency recording.
+
+### OutputCollector
+
+OutputCollector is a container that facilitates building up the native output format of Nighthawk (proto3, nighthawk::client::Output). It is the basis for all output formats offered by Nighthawk, including CLI human output.
+
+### OutputFormatter
+
+OutputFormatter is responsible for transformations of nighthawk::client::Output to requested formats (e.g. human, json, fortio, etc)
+
+### Statistic
+
+Nighthawk’s statistic is responsible for administrating latencies. The most notable implemenation that exists today wraps HdrHistogram, but Nighthawk also has a couple of other implementations which mostly exist to ensure that floating point math is correct in tests. For various reasons, HdrHistogram might get replaced by [libcirclhist](https://github.com/envoyproxy/nighthawk/issues/115) in the near future.
+
+### H1Pool & H2Pool
+
+Nighthawk derives its own version of these from the vanilla Envoy ones. It does that to implement things like pro-active connection prefetching and H2 multi-connection support, as well as offer more connection management strategies. 
+
+## Executables
+
+### nighthawk_client
+
+This is the CLI interface
+
+### nighthawk_service
+
+Nighthawk’s gRPC service is able to execute load tests, and also offers a stub request source gRPC service method implementation. 
+
+The stub request source gRPC method can be used for testing remote request source configuration (either with nighthawk_client, or by configuring request-source to loop back when requesting nighthawk_service to perform a load test). 
+
+While the stub gRPC method has no production use today, perhaps in the future this service method may get configurable so it can read traffic streams from disk, or from yet another remote source (which would allow the gRPC service to serve as a workload distributor for its Workers in a benchmark execution, as well as be convenient for those implementing a gRPC request source as they only have a single inbound gRPC call instead of one per worker).
```

Comment:
Why are we calling it a "stub gRPC method"? Or maybe I am unclear which method we are referring to. Is this the new method added in #229 ?

Secondly the language sounds fairly uncertain about its use. If this is about #229, we probably have a better understanding today as to how this will be used. It was already mentioned earlier in this doc as a reference to log-replay (non-static requests). Can we reword this accordingly? We can still keep the uncertain language about the fact that we may refactor this into a plugin-like architecture where we will have other request sources. Like loading from a file.